wcfsetup/install/files/ts/WoltLabSuite/Core/Media/Manager/Base.ts

```diff
@@ -0,0 +1,550 @@
+
+      if (selectedCategoryId) {
+        const newCategoryId = ~~media.categoryID;
+
+        if (
+          oldCategoryId != newCategoryId &&
+          (oldCategoryId == selectedCategoryId || newCategoryId == selectedCategoryId)
+        ) {
+          this._search!.search();
+        }
+      }
+    }
+
+    UiDialog.open(this);
+
+    this._media.set(~~media.mediaID, media);
+
+    const listItem = this._listItems.get(~~media.mediaID)!;
+    const p = listItem.querySelector(".mediaTitle")!;
+    if (media.isMultilingual) {
+      if (media.title && media.title[window.LANGUAGE_ID]) {
+        p.textContent = media.title[window.LANGUAGE_ID];
+      } else {
+        p.textContent = media.filename;
+      }
+    } else {
+      if (media.title && media.title[media.languageID!]) {
+        p.textContent = media.title[media.languageID!];
+      } else {
+        p.textContent = media.filename;
+      }
+    }
+
+    const thumbnail = listItem.querySelector(".mediaThumbnail")!;
+    thumbnail.innerHTML = media.elementTag;
+    // Bust browser cache by adding additional parameter.
+    const img = thumbnail.querySelector("img");
+    if (img) {
+      img.src += `&refresh=${Date.now()}`;
+    }
+  }
+
+  /**
+   * Initializes the dialog pagination.
+   */
+  protected _initPagination(pageCount: number, pageNo?: number): void {
+    if (pageNo === undefined) pageNo = 1;
+
+    if (pageCount > 1) {
+      const newPagination = document.createElement("div");
+      newPagination.className = "paginationBottom jsPagination";
+      DomUtil.replaceElement(
+        UiDialog.getDialog(this)!.content.querySelector(".jsPagination") as HTMLElement,
+        newPagination,
+      );
+
+      this._pagination = new UiPagination(newPagination, {
+        activePage: pageNo,
+        callbackSwitch: this._search!.search.bind(this._search),
+        maxPage: pageCount,
+      });
+    } else if (this._pagination) {
+      this._pagination.getElement().style.display = "none";
+    }
+  }
+
+  /**
+   * Removes all media clipboard checkboxes.
+   */
+  _removeClipboardCheckboxes(): void {
+    this._mediaManagerMediaList!.querySelectorAll(".mediaCheckbox").forEach((el) => el.remove());
+  }
+
+  /**
+   * Opens the media editor after uploading a single file.
+   *
+   * @since 5.2
+   */
+  _openEditorAfterUpload(data: MediaUploadSuccessEventData): void {
+    if (data.upload === this._upload && !data.isMultiFileUpload && !this._upload.hasPendingUploads()) {
+      const keys = Object.keys(data.media);
+
+      if (keys.length) {
+        UiDialog.close(this);
+
+        this._mediaEditor!.edit(this._media.get(~~data.media[keys[0]].mediaID)!);
+      }
+    }
+  }
+
+  /**
+   * Sets the displayed media (after a search).
+   */
+  _setMedia(media: object): void {
+    this._media = new Map<number, Media>();
+    Object.entries(media).forEach(([mediaId, media]) => {
+      this._media.set(~~mediaId, media);
+    });
```

Comment:
```suggestion
    this._media = new Map<number, Media>(Object.entries(media).map(([mediaId, media]) => [~~mediaId, media]));
```
I believe this should work.